frontend/app/screens/Sessions/index.js

```diff
@@ -0,0 +1,200 @@
+      setTrack(track);
+      setSessions(
+        sessions.map(el => {
+          if (el.track === track) {
+            return { ...el, visibility: true };
+          }
+          return { ...el, visibility: false };
+        }),
+      );
+    },
+    filterByText: ({ setSessions, sessions, setSearchText }) => ({
+      target: { value },
+    }) => {
+      setSearchText(value);
+      if (!value) {
+        setSessions(sessions.map(el => ({ ...el, visibility: true })));
+      }
+      const sessionsFilteredBySearchResults = searchArr(
+        value,
+        sessions,
+      ).map((isMatch, eachInd) => ({
+        ...sessions[eachInd],
+        visibility: isMatch,
+      }));
+      setSessions(sessionsFilteredBySearchResults);
+    },
+    resetAllFilters: ({
+      sessions,
+      setSessions,
+      setSkillLevel,
+      setTrack,
+      setSearchText,
+    }) => () => {
+      setSessions(sessions.map(el => ({ ...el, visibility: true })));
+      setSkillLevel(null);
+      setTrack(null);
+      setSearchText('');
+    },
+  }),
+  withProps(() => ({
+    tracks: [
+      'Coding and Development',
+      'Devops',
+      'Frontend',
+      'Horizons',
+      'PHP',
+      'Project Management',
+      'Site Building',
+      'UX and Content Strategy',
+    ],
+    skillLevels: ['Beginner', 'Intermediate', 'Advanced'],
+  })),
+  lifecycle({
+    componentWillMount() {
+      const { sessions, setSessions } = this.props;
+      setSessions(sessions.map(el => ({ ...el, visibility: true })));
+    },
+  }),
+);
+
+type SessionsListProps = {
+  sessions: Array<SessionExtendedT>,
+  filterBySkillLevel: Function,
+  resetAllFilters: Function,
+  filterByText: Function,
+  filterByTrack: Function,
+  tracks: Array<TrackT>,
+  skillLevels: Array<SkillLevelT>,
+  textSearchInput: string,
+};
+
+const SessionsList = withLogic((props: SessionsListProps) => {
+  const {
+    sessions,
+    filterBySkillLevel,
+    skillLevels,
+    tracks,
+    filterByTrack,
+    resetAllFilters,
+    filterByText,
+    textSearchInput,
+  } = props;
+  return (
+    <div className={styles.sessionsContainer}>
+      <h2>Skill Level Filters:</h2>
+      {skillLevels.map(eachLevel =>
+        (<button key={eachLevel} onClick={() => filterBySkillLevel(eachLevel)}>
+          {`${eachLevel}`}
+        </button>),
+      )}
+      <h2>Track Filters:</h2>
+      {tracks.map(eachTrack =>
+        (<button key={eachTrack} onClick={() => filterByTrack(eachTrack)}>
+          {`${eachTrack}`}
+        </button>),
+      )}
+      <div>
+        <h2>Session Search</h2>
+        <input type="text" value={textSearchInput} onChange={filterByText} />
+      </div>
+      <h2>Reset All:</h2>
+      <button onClick={() => resetAllFilters()}>Reset All Filters</button>
+      <hr />
+      {sessions
+        .filter(({ visibility }) => visibility === true)
+        .map(eachSession =>
+          (<div key={eachSession.title}>
```

Comment:
lets reduce the file size by using more components. This "SessionTeaser?" is an easy one, but maybe also all the logic stuff can go in it's own file